### PR TITLE
Add `maxLength` option

### PR DESCRIPTION
**Pull request by e-cloud**
it's too limited to has only hardcoded MAX_FILENAME_LENGTH.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Duplicate of https://github.com/sindresorhus/filenamify/issues/5

What's your use-case for needing more than 100 characters?

**Comment by scott-ho** [comment]
That depends on requirement. For one case, if someone need to log something into the file based on a java package name, it could be very long. 'xxxxxxxxxxx.xxxxxxxxxx.xxxxxxxxxxx.xxxxxxxxxxxxxx.xxxxxxxxxxx.xxxxxxxxxx.xxxxxxxxxx.xxxxxxxxxxx.xxxxxxxxxxx.xxxxxxxxxx.xxxClass'.

For another, `وزير الخارجية النمساوي: الوضع فى ليبيا "لا يزال بالغ السكك الحديدية في` is a news title of Arab language.

I don't recommend hardcoding the constraints

**Comment by sindresorhus** [comment]
@scott-ho Alright

**Comment by sindresorhus** [comment]
@e-cloud I think the option should be named `maxLength`. And it needs a test.

File: index.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ const filenamify = (string, options = {}) => {
 	}
 
 	string = filenameReservedRegex.windowsNames().test(string) ? string + replacement : string;
-	string = string.slice(0, MAX_FILENAME_LENGTH);
+	string = string.slice(0, options.maxLength ? options.maxLength : MAX_FILENAME_LENGTH);
```
Comment (sindresorhus) [inline index.js]:
```suggestion
	string = string.slice(0, typeof options.maxLength === 'number' ? options.maxLength : MAX_FILENAME_LENGTH);
```

**Comment by sindresorhus** [comment]
>  think the option should be named maxLength.

You didn't update all the places, the PR title included.

**Comment by sindresorhus** [comment]
Still not. Please look at the diff.

**Pull request merged**